Classes/git/PBGitGrapher.mm

```diff
@@ -46,22 +46,33 @@ - (id) initWithRepository: (PBGitRepository*) repo
 	return self;
 }
 
-void add_line(struct PBGitGraphLine *lines, int *nLines, int upper, int from, int to, int index)
+int long_to_integer_heuristic(long input)
+{
+	if (input < INT_MIN) {
+		return INT_MIN;
+	} else if (input > INT_MAX) {
+		return INT_MAX;
+	}
+	return (int)input;
+}
```

Comment:
These bounds are too big; they need to match the bitfield size to saturate correctly: http://codepad.org/K6plCqhp